cli/npm/installer/common/deno_task_executor.rs

```diff
@@ -0,0 +1,337 @@
+// Copyright 2018-2025 the Deno authors. MIT license.
+
+use std::borrow::Cow;
+use std::collections::HashSet;
+use std::path::PathBuf;
+use std::rc::Rc;
+
+use deno_core::error::AnyError;
+use deno_npm::resolution::NpmResolutionSnapshot;
+use deno_npm::NpmResolutionPackage;
+use deno_resolver::npm::ManagedNpmResolverRc;
+use deno_runtime::deno_io::FromRawIoHandle;
+use deno_task_shell::KillSignal;
+
+use super::bin_entries::BinEntries;
+use super::lifecycle_scripts::is_broken_default_install_script;
+use super::lifecycle_scripts::LifecycleScriptsExecutor;
+use super::lifecycle_scripts::LifecycleScriptsExecutorOptions;
+use super::lifecycle_scripts::PackageWithScript;
+use super::lifecycle_scripts::LIFECYCLE_SCRIPTS_RUNNING_ENV_VAR;
+use super::CachedNpmPackageExtraInfoProvider;
+use super::ExpectedExtraInfo;
+use crate::sys::CliSys;
+use crate::task_runner::TaskStdio;
+use crate::util::progress_bar::ProgressMessagePrompt;
+
+#[derive(Debug, thiserror::Error, deno_error::JsError)]
+pub enum DenoTaskLifecycleScriptsError {
+  #[class(inherit)]
+  #[error(transparent)]
+  Io(#[from] std::io::Error),
+  #[class(inherit)]
+  #[error(transparent)]
+  BinEntries(#[from] super::bin_entries::BinEntriesError),
+  #[class(inherit)]
+  #[error(
+    "failed to create npm process state tempfile for running lifecycle scripts"
+  )]
+  CreateNpmProcessState(#[source] std::io::Error),
+  #[class(generic)]
+  #[error(transparent)]
+  Task(AnyError),
+  #[class(generic)]
+  #[error("failed to run scripts for packages: {}", .0.join(", "))]
+  RunScripts(Vec<String>),
+}
+
+pub struct DenoTaskLifeCycleScriptsExecutor {
+  npm_resolver: ManagedNpmResolverRc<CliSys>,
+}
+
+#[async_trait::async_trait(?Send)]
+impl LifecycleScriptsExecutor for DenoTaskLifeCycleScriptsExecutor {
+  async fn execute(
+    &self,
+    options: LifecycleScriptsExecutorOptions<'_>,
+  ) -> Result<(), AnyError> {
```

Comment:
The code inside this method is mostly the same as before, except now it uses the `options` object.